layer/frame_time_layer.cc

```diff
@@ -117,6 +146,23 @@ void CreateFinishIndicatorFile(const char* finishCause) {
   fclose(finish_file);
 }
 
+bool FrameTimeLayerData::HasBenchmarkStarted() {
+  if (benchmark_state_idx_ > 0 || benchmark_start_pattern_.empty()) return true;
+
+  if (!benchmark_log_scanner_.has_value() || !benchmark_log_scanner_->is_open())
+    return true;
+
+  if (benchmark_log_scanner_->consume_new_lines()) {
+    benchmark_state_idx_ = 1;
+    benchmark_log_scanner_->get_first_occurrence_line_num(
+        benchmark_start_pattern_);
+    benchmark_log_scanner_->close();
```

Comment:
IMHO exposing close() and is_open() pollutes the LogScanner API - the layer could just clear the optional<> when it's done with the scanner (see my other comment about simplifying the LogScanner API)

Reply:
Good suggestion, I like this idea